.Lib9c.Tools/SubCommand/State.cs

```diff
@@ -238,6 +238,50 @@ public void Set(IDictionary<byte[], byte[]> values)
 
             public bool Exists(byte[] key) => _dictionary.ContainsKey(key);
 
+            public byte[] Get(in KeyBytes key) => _dictionary[key.ToByteArray()];
+
+            public IReadOnlyDictionary<KeyBytes, byte[]> Get(IEnumerable<KeyBytes> keys)
+            {
+                var dictBuilder = ImmutableDictionary.CreateBuilder<KeyBytes, byte[]>();
+                foreach (KeyBytes key in keys)
+                {
+                    if (_dictionary.TryGetValue(key.ToByteArray(), out var value) && value is { } v)
+                    {
+                        dictBuilder[key] = v;
+                    }
+                }
+
+                return dictBuilder.ToImmutable();
+            }
+
+            public void Set(in KeyBytes key, byte[] value) =>
+                _dictionary[key.ToByteArray()] = value;
+
+            public void Set(IDictionary<KeyBytes, byte[]> values)
+            {
+                foreach (KeyValuePair<KeyBytes, byte[]> kv in values)
+                {
+                    _dictionary[kv.Key.ToByteArray()] = kv.Value;
+                }
+            }
+
+            public void Delete(in KeyBytes key) =>
+                _dictionary.TryRemove(key.ToByteArray(), out _);
+
+            public void Delete(IEnumerable<KeyBytes> keys)
+            {
+                foreach (KeyBytes key in keys)
+                {
+                    _dictionary.TryRemove(key.ToByteArray(), out _);
+                }
+            }
+
+            public bool Exists(in KeyBytes key) =>
+                _dictionary.ContainsKey(key.ToByteArray());
+
+            IEnumerable<KeyBytes> IKeyValueStore.ListKeys() =>
+                (IEnumerable<KeyBytes>)_dictionary.Keys;
```

Comment:
Existed methods with the same name still remains above.  These methods implemented the previous version of `IKeyValueStore` interface, but now they are unnecessary as they no more satisfy the current interface.

The changes made on `IKeyValueStore` from Libplanet 0.24 to 0.25 are mainly about key types.  Previously, the interface represented keys as `byte[]`, but now they are represented in `KeyBytes`.

In .NET, arrays do not implement `GetHashCode()` or any methods to determine if two arrays consist of the same elements with the same order.  There's a common mistake to use arrays as dictionary keys, because it causes a dictionary to allow to have duplicate keys.  In order to correctly use arrays as dictionary keys, we should implement an `IEqualityComparer<T[]>` and let dictionaries use it for comparing keys.  That's why `ArrayEqualityComparer<T>` class exists inside this `MemoryKeyValueStore` class.

`KeyBytes` was introduced to prevent from such mistake.  It implements `GetHashCode()` and `IEquatable<KeyBytes>` so that it can be dictionary keys without any burden.  It means we can safely delete `ArrayEqualityComparer<T>` as well if we change the type of `_dictionary` from `ConcurrentDictionary<byte[], byte[]>` to `ConcurrentDictionary<KeyBytes, byte[]>`.

To sum up:

- `_dictionary` now can be `ConcurrentDictionary<KeyBytes, byte[]>` and IMHO it's much better.
- Existed methods are now unnecessary.  Let's get rid of them.
- `ArrayEqualityComparer<T>` class is also unnecessary.  Let's remove it!